staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -170,6 +187,20 @@ func TestWatchErrorIncorrectConfiguration(t *testing.T) {
 			requestOpts: storage.ListOptions{ProgressNotify: true},
 			expectedErr: apierrors.NewInternalError(errors.New("progressNotify for watch is unsupported by the etcd storage because no newFunc was provided")),
 		},
+		{
+			name:            "sendInitialEvents and no newFunc provided",
+			enableWatchList: true,
+			setupFn:         func(opts *setupOptions) { opts.newFunc = nil },
+			requestOpts:     storage.ListOptions{SendInitialEvents: ptr.To(true), Predicate: storage.SelectionPredicate{Field: fields.Everything(), Label: labels.Everything(), AllowWatchBookmarks: true}},
+			expectedErr:     apierrors.NewInvalid(schema.GroupKind{Group: "", Kind: "pods"}, "", field.ErrorList{field.Forbidden(field.NewPath("sendInitialEvents"), "for watch is unsupported by the etcd storage because no newFunc was provided")}),
+		},
+		{
+			name:            "sendInitialEvents and no getCurrentResourceVersionFromStorage provided",
+			enableWatchList: true,
+			setupFn:         func(opts *setupOptions) { opts.newListFunc = nil },
+			requestOpts:     storage.ListOptions{SendInitialEvents: ptr.To(false), Predicate: storage.SelectionPredicate{Field: fields.Everything(), Label: labels.Everything(), AllowWatchBookmarks: true}},
+			expectedErr:     fmt.Errorf("failed to get the current resource version from the storage: newListFunction wasn't provided for *example.Pod"),
```

Comment:
note that here we don't return an api error. This is because getCurrentRVFromStorage calls directly the underlying function. We could change that and before calling the underlying function check `newListFunction` and return an api error.

Reply:
Yeah - we should fix that, we want to return only api errors.